sdk/identity/identity/Troubleshooting.md

```diff
@@ -0,0 +1,241 @@
+## Troubleshooting Azure Identity Authentication Issues
+
+The Azure Identity SDK offers various `TokenCredential` implementations. These implementations tend to throw `CredentialUnavailableError` and `AuthenticationRequired` exceptions.
+The `CredentialUnavailableError` indicates that the credential cannot execute in the current environment setup due to lack of required configuration.
+The `AuthenticationRequiredError` indicates that the credential was able to run/execute but ran into an authentication issue from the server's end. This can happen due to invalid configuration/details passed in to the credential at construction time.
+This troubleshooting guide covers mitigation steps to resolve these exceptions thrown by various `TokenCredential` implementations in the Azure Identity Typescript client library.
+
+## Table of contents
+
+- [Troubleshooting Default Azure Credential Authentication Issues](#troubleshooting-default-azure-credential-authentication-issues)
+- [Troubleshooting Application Credential Authentication Issues](#troubleshooting-application-credential-authentication-issues)
+- [Troubleshooting Environment Credential Authentication Issues](#troubleshooting-environment-credential-authentication-issues)
+- [Troubleshooting Service Principal Authentication Issues](#troubleshooting-service-principal-authentication-issues)
+- [Troubleshooting Username Password Authentication Issues](#troubleshooting-username-password-authentication-issues)
+- [Troubleshooting Mananged Identity Authentication Issues](#troubleshooting-mananged-identity-authentication-issues)
+- [Troubleshooting Visual Studio Code Authentication Issues](#troubleshooting-visual-studio-code-authentication-issues)
+- [Troubleshooting Azure CLI Authentication Issues](#troubleshooting-azure-cli-authentication-issues)
+- [Troubleshooting Azure Powershell Authentication Issues](#troubleshooting-azure-powershell-authentication-issues)
+
+## Troubleshooting Default Azure Credential Authentication Issues
+
+### Credential Unavailable Error
+
+The `DefaultAzureCredential` attempts to retrieve an access token by sequentially invoking a chain of credentials. The `CredentialUnavailableError` in this scenario signifies that all the credentials in the chain failed to retrieve the token in the current environment setup/configuration. You need to follow the configurtion instructions for the respective credential you're looking to use via `DefaultAzureCredential` chain, so that the credential can work in your environment.
```

Comment:
It would be great to talk about how to see the DefaultAzureCredential logs, to understand what failed. Perhaps to show examples of how these logs would look.

Reply:
so should i pick example from one of our tests?

Reply:
No no, try to use the DefaultAzureCredential locally with the AZURE_LOG_LEVEL set to verbose and put here how those logs look, perhaps with a small paragraph of how to read them.